supplier_app/tests/test_view.py

```diff
@@ -1601,3 +1603,173 @@ def test_taxpayer_creation_error_should_display_error_message(self):
+        
+        ADDRESS_POST = {
+            'address_form-street': 'San Martin',
+            'address_form-number': '21312',
+            'address_form-zip_code': '123',
+            'address_form-city': 'Mendoza',
+            'address_form-state': 'Mendoza',
+            'address_form-country': 'Argentina',
+        }
+
+        row_before_modification = len(address.history.all())
+
+        self.client.post(
+            reverse(
+                'address-update',
+                kwargs=kwargs
+            ),
+            data=ADDRESS_POST
+        )
+
+        row_after_modification = len(address.history.all())
+
+        last_row_address_history = address.history.last()
+
+        self.assertGreater(row_after_modification, row_before_modification)
+        self.assertEqual(address.street, last_row_address_history.street)
+        self.assertEqual(address.number, last_row_address_history.number)
+        self.assertEqual(address.zip_code, last_row_address_history.zip_code)
+        self.assertEqual(address.city, last_row_address_history.city)
+        self.assertEqual(address.state, last_row_address_history.state)
+        self.assertEqual(address.country, last_row_address_history.country)
+
+    def test_create_new_row_in_history_bank_account_table(self):
+        bank_account = BankAccountFactory(taxpayer=self.taxpayer)
+        kwargs = {
+            'taxpayer_id': self.taxpayer.id,
+            'bank_id': bank_account.id,
+        }
+
+        BANK_ACCOUNT_POST = {
+            'bank_info': get_bank_info_example("CITIBANK N.A."),
+            'bank_account_number': '123214',
+        }
+
+        row_before_modification = len(bank_account.history.all())
+
+        self.client.post(
+            reverse(
+                'bank-account-update',
+                kwargs=kwargs
+            ),
+            data=BANK_ACCOUNT_POST
+        )
+
+        row_after_modification = len(bank_account.history.all())
+
+        last_row_bank_account_history = bank_account.history.last()
+
+        self.assertGreater(row_after_modification, row_before_modification)
+        self.assertEqual(bank_account.bank_account_number, last_row_bank_account_history.bank_account_number)
+        self.assertEqual(bank_account.bank_info, last_row_bank_account_history.bank_info)
+
+    def test_history_taxpayer_should_contain_old_and_new_values(self):
+        old_name = self.taxpayer.business_name
+        self.taxpayer.business_name = "New_SupraSA"
+        new_name = self.taxpayer.business_name
+        self.taxpayer.save()
+
+        response = self.client.get(
+            reverse(
+                'taxpayer-history',
+                kwargs={'pk': self.taxpayer.id}
+                ),
+        )
+
+        self.assertContains(response, old_name)
+        self.assertContains(response, new_name)
+        self.assertEqual(response.template_name[0], 'AP_app/taxpayer-history-list.html')
+
+    def test_history_address_should_contain_old_and_new_values(self):
```

Comment:
Seems like we are testing storing all results, but without a way to identify what's new and what's old. Could we check this?